packages/rs-drive-abci/src/query/mod.rs

```diff
@@ -441,6 +443,98 @@ impl<C> Platform<C> {
                 };
                 Ok(QueryValidationResult::new_with_data(response_data))
             }
+            "/dataContractHistory" => {
+                let GetDataContractHistoryRequest {
+                    id,
+                    limit,
+                    offset,
+                    start_at_seconds,
+                    prove,
+                } = check_validation_result_with_data!(GetDataContractHistoryRequest::decode(
+                    query_data
+                ));
+                let contract_id: Identifier = check_validation_result_with_data!(id.try_into());
+
+                // TODO: make a cast safe
+                let limit = limit
+                    .map(|limit| {
+                        u16::try_from(limit).map_err(|_| {
+                            Error::Drive(drive::error::Error::Contract(ContractError::Overflow(
+                                "can't fit u16 limit from the supplied value",
+                            )))
+                        })
+                    })
+                    .transpose()?;
```

Comment:
Would be good to see what happens if someone sends a request with a limit  above u16:Max.